### PR TITLE
[RLlib] ConnectorV2 API: Add heuristic action logits mixin example script.

### PR DESCRIPTION
**Pull request by sven1977**
ConnectorV2 API: Add prev-a/prev-r and framestacking example scripts to CI tests.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/examples/connectors/connector_v2_frame_stacking.py [hunk 1]
```diff
@@ -86,27 +90,14 @@ def _make_learner_connector(input_observation_space, input_action_space):
...
-                                            **dict(cfg, **{"render_mode": "rgb_array"})
-                                        )
-                                    )
-                                )
-                            )
-                        )
-                    )
-                )
+        lambda cfg: (
+            wrap_atari_for_new_api_stack(
```
Comment (simonsays1980) [inline rllib/examples/connectors/connector_v2_frame_stacking.py:94]: IsnÄt there a gym.Wrapper for framestacking?

File: rllib/examples/connectors/connector_v2_prev_rewards_actions_inputs.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+            ),
+        )
+        .resources(num_learner_workers=0)
+        .training(
+            learner_connector=_learner_connector,
+            num_sgd_iter=6,
+            vf_loss_coeff=0.01,
+            train_batch_size=4000,
+            model=dict({
+                "use_lstm": True,
```
Comment (simonsays1980) [inline rllib/examples/connectors/connector_v2_prev_rewards_actions_inputs.py:103]: Don't we need a corresponding `max_seq_len`?

**Review by simonsays1980 — APPROVED** [review]
LGTM. LSTMs and GRUs (probably also Attention networks) should now be possible with these connectors.

File: rllib/connectors/common/prev_action_prev_reward.py [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+        # TODO (sven): Refine this new space computation here. We need a better utllty
+        #  to compute the dimensions of a flattened space.
+        #  For now, this only works on Discrete action spaces and 1D Box observation
+        #  spaces.
+        return gym.spaces.Box(
+            float("-inf"),
+            float("inf"),
+            shape=(
+                self.input_observation_space.shape[0]
+                + self.input_action_space.n * self.n_prev_actions
```
[the remark this answers is not in the document]
Comment (sven1977) [reply rllib/connectors/common/prev_action_prev_reward.py:31]: No, it's not. Yeah, this is a restriction of this example connector. I'll try to generalize this. It's gonna bite us back if we don't, I think :)

File: rllib/connectors/common/prev_action_prev_reward.py [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+                        batch=data,
+                        column=SampleBatch.PREV_REWARDS,
+                        single_agent_episode=sa_episode,
+                        item_to_add=prev_n_rewards,
+                    )
+
+        return data
+
+    def _build_inference_data(self, episodes, data):
+        for sa_episode in self.single_agent_episode_iterator(episodes):
```
[the remark this answers is not in the document]
Comment (sven1977) [reply rllib/connectors/common/prev_action_prev_reward.py:134]: Correct, there are multiple episodes, if and only if there is vectorization. Thus: `len(episodes) == config.num_envs_per_worker`.

File: rllib/connectors/common/prev_action_prev_reward.py [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+                    item_to_add=prev_n_actions,
+                )
+
+            if self.n_prev_rewards:
+                self.add_batch_item(
+                    batch=data,
+                    column=SampleBatch.PREV_REWARDS,
+                    single_agent_episode=sa_episode,
+                    item_to_add=np.array(
+                        sa_episode.get_rewards(
```
[the remark this answers is not in the document]
Comment (sven1977) [reply rllib/connectors/common/prev_action_prev_reward.py:158]: Thanks! Love it, too. No more stupid buffers that have to be reset (and filled with something zero'y) when the env just got reset. :)

File: rllib/connectors/learner/prev_action_prev_reward.py [hunk 6]
```diff
@@ -0,0 +1,10 @@
+from functools import partial
+
+from ray.rllib.connectors.common.prev_action_prev_reward import (
+    _PrevRewardPrevActionConnector,
+)
+
+
+PrevRewardPrevActionLearner = partial(
```
[the remark this answers is not in the document]
Comment (sven1977) [reply rllib/connectors/learner/prev_action_prev_reward.py:8]: The `PrevRewardPrevActionLearner` could be used in the learner pipeline.

There are two competing designs right now (and I haven't really figured out the final solution, which I think is ok, though, user can decide):
* Use one env-to-module and one learner connector, both just grabbing the information from the episodes and writing it to their respective batches (action batch and learner batch). But neither of them writes the changed information back to the episodes (permanently).
* Only one env-to-module connector (no learner connector). But here, the env-to-module connector grabs the information from the episode, puts it into the batch, and then writes the transformed information back into the episode, thus making it permanent (so the learner connector doesn't have to do anything anymore).

...

There are cases, in which the first one - albeit a little more complex b/c of the two connectors - is faster than the other, e.g. Atari framestacking.

File: rllib/core/models/torch/encoder.py [hunk 7]
```diff
@@ -329,16 +329,22 @@ def __init__(self, config: RecurrentEncoderConfig) -> None:
...
-        if lstm_weights_initializer:
-            lstm_weights_initializer(
-                self.lstm.all_weights, **config.hidden_weights_initializer_config or {}
-            )
-        # Initialize LSTM layer bias, if necessary.
-        if lstm_bias_initializer:
-            lstm_bias_initializer(
-                self.lstm.bias, **config.hidden_bias_initializer_config or {}
-            )
+        # Initialize LSTM layer weigths and biases, if necessary.
```
[the remark this answers is not in the document]
Comment (sven1977) [reply rllib/core/models/torch/encoder.py:332]: Yeah, agreed. No worries, I had to actually debug into the code to inspect the LSTM torch layer object to figure this one out. The documentation is bad in this case.

File: rllib/examples/connectors/connector_v2_frame_stacking.py (hunk 1 above)
Comment (sven1977) [reply rllib/examples/connectors/connector_v2_frame_stacking.py:94]: There is, but this would blow up the memory taken by the observations sometimes by n-frames.
This is just an example. I could actually show that using two connectors here (instead of such an env wrapper): learner AND env-to-module actually speeds up Atari runs by 5% :)

File: rllib/BUILD [hunk 8]
```diff
@@ -2851,31 +2851,49 @@ py_test(
...
+    args = ["--enable-new-api-stack", "--num-agents=2", "--as-test", "--stop-reward=800.0", "--framework=torch", "--algo=PPO"]
+)
+
 # ----------------------
 # old connector examples
 # ----------------------
 py_test(
-    name = "examples/connectors/run_connector_policy",
-    main = "examples/connectors/run_connector_policy.py",
+    name = "examples/connectors/v1/run_connector_policy",
```
Comment (sven1977) [inline rllib/BUILD:2876]: Moved old connector examples into sub folder to reduce confusion.

File: rllib/connectors/input_output_types.py [hunk 9]
```diff
@@ -1,75 +0,0 @@
-from enum import Enum
```
Comment (sven1977) [inline rllib/connectors/input_output_types.py:1]: This is no longer needed.
We now provide some light-weight utility methods within the ConnectorV2 API to simplify batch building, if desired by the users. However, they can also always just manually manipulate the batch (it's always just a dict), if they want to.

File: rllib/connectors/learner/prev_action_prev_reward.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (simonsays1980) [reply rllib/connectors/learner/prev_action_prev_reward.py]: I see. Isn't possible to work with views in such scenarios? The memory of numpy arrays is usually not duplicated
